### PR TITLE
DEV: work around pathlib bug affecting dev.py for Python 3.9 on Windows

### PR DESCRIPTION
**Pull request by rgommers**
Also remove a workaround in a Windows job for this issue, and clean up the code for `dev.py --win-cp-openblas` some more.

Note that `--win-cp-openblas` still doesn't always work as designed - in particular, `openblas_support.py` will silently fail to do the right thing on Windows if the git repo checkout is on the D: drive while OpenBLAS gets downloaded to the C: drive. That is happening in a GHA CI job, and it's plagued us on Azure before too (see gh-11967). It's nontrivial to fix (the `C:` drive is hardcoded in multiple scripts), so I have just added a comment to explain why the two lines in the CI job that really shouldn't have to be there are necessary.

The main win here is that `--build-dir` and `--install-prefix` are no longer needed.

### CODE REVIEW

File: dev.py [hunk 1]
```diff
@@ -609,6 +618,7 @@ def copy_openblas(cls, dirs):
         # Look in bin subdirectory for OpenBLAS binaries.
         bin_path = openblas_lib_path.parent / 'bin'
         # Locate, make output .libs directory in Scipy install directory.
+        print(dirs.site)
```
Comment (andyfaff) [inline dev.py:621]: Remove this print statement?

File: dev.py [hunk 2]
```diff
@@ -306,14 +306,22 @@ def __init__(self, args=None):
...
+
+        if sys.platform == 'win32' and sys.version_info < (3, 10):
+            # Work around a pathlib bug; these must be absolute paths
+            self.build = Path(os.path.abspath(self.build))
+            self.installed = Path(os.path.abspath(self.installed))
+
         # relative path for site-package with py version
         # i.e. 'lib/python3.10/site-packages'
         self.site = self.get_site_packages()
+        print(self.site)
```
Comment (andyfaff) [inline dev.py:324]: Remove this print statement?

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -112,11 +112,10 @@ jobs:
...
         run: |
-          # TODO: remove the build-dir and install-prefix flags once dev.py
-          # supplies meson with absolute paths. This is due to a pathlib bug
-          # on cp39-win.
-          python dev.py --build-dir=$pwd\build --install-prefix=$pwd\build-install build -j 2 --win-cp-openblas
-          meson install -C build
+          python dev.py build -j 2 --win-cp-openblas
+          # Necessary because GitHub Actions checks out the repo to D:\ while OpenBLAS
+          # got installed to C:\ higher up. The copying with `--win-cp-openblas` fails
+          # when things are split over drives.
```
Comment (andyfaff) [inline .github/workflows/windows.yml:118]: What's the ultimate reason copying isn't possible between different drives?

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -112,11 +112,10 @@ jobs:
...
           cp C:\opt\64\bin\*.dll $pwd\build-install\Lib\site-packages\scipy\.libs\
```
Comment (andyfaff) [inline .github/workflows/windows.yml:119]: Can you apply similar changes to L61-L70?

**Review by andyfaff — APPROVED** [review]
A few nits. Otherwise LGTM.

The lint job is failing, it's in a totally different location, but there you go.

File: .github/workflows/windows.yml (hunk 3 above)
Comment (rgommers) [reply .github/workflows/windows.yml]: Windows design, as far as I can tell. Unix-like systems have a single filesystem starting from `/` with mount points. Windows drives are really separate, no common `/`. Windows Explorer and GUI tools know how to deal with that, Python and Bash CLI tools don't.

File: .github/workflows/windows.yml (hunk 4 above)
Comment (rgommers) [reply .github/workflows/windows.yml:109]: Done, with a few more changes to keep the two jobs in sync. We may want to merge them later perhaps, they're almost the same.

**Comment by rgommers** [comment]
> The lint job is failing, it's in a totally different location, but there you go.

The complaint is
```
dev.py:849:23: UP031 [*] Use format specifiers instead of percent format
```
for this code:
```python
print("Running benchmarks for Scipy version %s at %s"
      % (version, mod_path))
```
We have a ton of code that uses such format specifiers and that's perfectly valid. I'll open a bug report, it doesn't seem okay for the linter to choke on that.

This PR should be good to merge.

**Comment by j-bowhay** [comment]
> We have a ton of code that uses such format specifiers and that's perfectly valid. I'll open a bug report, it doesn't seem okay for the linter to choke on that.

It is because currently the pyupgrade rules are enabled, this flags "old" style syntax

**Pull request merged**